### ISSUE TITLE
Fedora 35 package

### ISSUE THREAD

**Issue by muety** [Azure/azure-storage-fuse#684]
I'm on Fedora 35 and would like to use `blobfuse` there. Unfortunately, it looks like no package exists and the RPMs for RHEL and CentOS don't work (`nothing provides libboost_filesystem.so.1.66.0()(64bit) needed by blobfuse-1.4.3-1.el8.x86_64`).

**Comment by vibhansa-msft** [comment]
We have not tried to build on fedora so far. Above error shows libboost is not available in your system. If you are not able to find it from standard Fedora packages then may be you need to install it manually.

**Comment by muety** [comment]
Apparently boost is already installed, probably a version mismatch?

```
Package boost-devel-1.76.0-4.fc35.x86_64 is already installed.
```

**Comment by vibhansa-msft** [comment]
Can you correct the version and try. On Ubn for local build we generally install below dependencies and it works


 sudo apt-get install pkg-config cmake libcurl4-gnutls-dev libgnutls28-dev uuid-dev libgcrypt20-dev libboost-all-dev gcc g++ libfuse-dev -y

**Comment by muety** [comment]
Boost version 1.66.0 seems to be quite old is not yet available in Fedora 35's package repository anymore. Would be nice if an official RPM for blobfuse could be published.

**Comment by vibhansa-msft** [comment]
So if you have latest boost installed, is it not compiling successfully for you ?

**Comment by muety** [comment]
No, with `boost-1.76.0-4` and `boost-devel-1.76.0-4` installed I still get the above error message.

**Comment by muety** [comment]
When trying to compile it from source, I get this message, which perhaps relates to the same issue.

```
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Boost (missing: filesystem system) (found version "1.76.0")
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake/Modules/FindBoost.cmake:2375 (find_package_handle_standard_args)
  CMakeLists.txt:87 (find_package)
```

**Comment by vibhansa-msft** [comment]
Line 87 in CMakeLists.txt is "find_package(Boost COMPONENTS filesystem system REQUIRED)" so build system is looking for "filesystem" and "system" components of Boost which its not able to find. Just for the trial if you comment out this line and see if we get any compilation failure at later stage.

**Comment by vibhansa-msft** [comment]
in CMakeLists.txt file there is a line commented saying Boost_USE_STATIC_LIBS_ON you can try uncommenting that one as well and keep the filesystem line as is.

**Comment by muety** [comment]
> Line 87 in CMakeLists.txt is "find_package(Boost COMPONENTS filesystem system REQUIRED)" so build system is looking for "filesystem" and "system" components of Boost which its not able to find. Just for the trial if you comment out this line and see if we get any compilation failure at later stage.

Yields this error:
```
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Boost (missing: thread) (found version "1.76.0")
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake/Modules/FindBoost.cmake:2375 (find_package_handle_standard_args)
  CMakeLists.txt:88 (find_package)
```

> in CMakeLists.txt file there is a line commented saying Boost_USE_STATIC_LIBS_ON you can try uncommenting that one as well and keep the filesystem line as is.

Compiles successfully, but raises runtime error when trying to execute:
```
blobfuse: error while loading shared libraries: libboost_filesystem.so.1.75.0: cannot open shared object file: No such file or directory
```

**Comment by vibhansa-msft** [comment]
I guess we need to manually install boost1.75 with filesystem. You can download tar from below location for required version:
https://www.boost.org/users/history/

Once you have the tar file you can follow below steps and hopefully this shall resolve the issue:
--------
tar zxf boost_1_75_0.tar.gz
cd boost_1_75_0
./bootstrap.sh --with-libraries=filesystem
./b2
sudo cp stage/lib/* /usr/local/lib64
ldconfig -n -v /usr/local/lib64
-----------

**Comment by vibhansa-msft** [comment]
Did it work for you ?

**Comment by muety** [comment]
Worked, thanks! :+1: 

Official Fedora RPM package would still be nice, of course, to simplify things.

**Comment by vibhansa-msft** [comment]
Sure we will look into putting official release for Fedora as well. Have added this to our backlog. Closing this issue here.
